teste2e/test_basic_vpc.py

```diff
@@ -11,14 +12,30 @@ def setUp(self):
         self.cluster = k8sCluster()
         self.api = k8sApi()
         vpc_name = "vpc1"
+        vpc_name_with_empty_vni = "vpc2"
+        vpc_name_with_dup_vni = "vpc3"
+        vni = 10
         subnet_name = "net1"
         subnet2_name = "net2"
         self.vpc = self.api.create_vpc(vpc_name, "12.0.0.0", "8")
+        
+        # Test vpc creation with empty vni
+        self.api.create_vpc(vpc_name_with_empty_vni, "13.0.0.0", "8", 1, None)
+        vpc_with_empty_vni = self.api.get_vpc_with_status_timeout(vpc_name_with_empty_vni)
+        self.assertNotEqual(vpc_with_empty_vni["vni"], None, "The vni shall not be none.")
+        self.assertEqual(vpc_with_empty_vni["status"], OBJ_STATUS.obj_provisioned, "The status shall be provisioned.")
+
+        # Test vpc creation with duplicate vni
+        self.vpc_with_dup_vni = self.api.create_vpc(vpc_name_with_dup_vni, "14.0.0.0", "8", 1, vpc_with_empty_vni["vni"])
+        vpc_with_dup_vni = self.api.get_vpc_with_status_timeout(vpc_name_with_dup_vni, OBJ_STATUS.vpc_status_duplicate_vni_error)
+        self.assertEqual(vpc_with_dup_vni["status"], OBJ_STATUS.vpc_status_duplicate_vni_error, "The status shall be duplicate vni error.")
+
```

Comment:
Please create a separate file for this test.

Reply:
Created